### PR TITLE
buildscripts: build macos artifacts in kokoro

### PR DESCRIPTION
**Pull request by zpencer**
This is the OSS changes, we need to also update the internal job definitions to run this.

### CODE REVIEW

File: buildscripts/kokoro/macos_artifacts.cfg [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Config file for internal CI
+
+# Location of the continuous shell script in repository.
+build_file: "grpc-java/buildscripts/kokoro/unix.sh"
```
Comment (ejona86) [inline buildscripts/kokoro/macos_artifacts.cfg]: macos_artifacts.sh?

File: buildscripts/make_dependencies.sh [hunk 2]
```diff
@@ -18,15 +22,17 @@ fi
 
 # Make protoc
 # Can't check for presence of directory as cache auto-creates it.
-if [ -f ${INSTALL_DIR}/bin/protoc ]; then
+if [[ -f ${INSTALL_DIR}/bin/protoc ]]; then
   echo "Not building protobuf. Already built"
 # TODO(ejona): swap to `brew install --devel protobuf` once it is up-to-date
 else
-  wget -O - https://github.com/google/protobuf/archive/v${PROTOBUF_VERSION}.tar.gz | tar xz -C $DOWNLOAD_DIR
+  if [[ ! -f "$DOWNLOAD_DIR/protobof-${PROTOBUF_VERSION}" ]]; then
```
Comment (ejona86) [inline buildscripts/make_dependencies.sh]: protobuf is misspelled.

File: buildscripts/kokoro/unix.sh [hunk 3]
```diff
@@ -36,3 +37,27 @@ pushd examples
...
+# Build release artifacts and move it to the artifacts dir
+# If we want to build but not store the artifacts, do not set 'define_artifacts' in cfg file.
+if [[ -z "${BUILD_MVN_ARTIFACTS:-}" ]]; then
+  exit 0
+fi
+
+# TODO(zpencer): if linux builds use this script, then also build for 32bit (-m32 flag)
+# Today, only macos uses this script and macos targets 64bit only
+
+# re-build protobuf using the flags listed in RELEASING.md
```
Comment (ejona86) [inline buildscripts/kokoro/unix.sh]: This script seems substantially different. It seems this should be a completely different script. We don't need those gradle properties nor examples building. We just need the compiler.

Comment (zpencer) [reply buildscripts/kokoro/unix.sh]: I wanted to make sure the build succeeds before trying to build the artifacts. I think I will call out to the original script before calling this one to achieve that.

**Comment by zpencer** [comment]
Also made some minor adjustments so that the script can be run easily on our workstations.

File: buildscripts/kokoro/unix.sh [hunk 4]
```diff
@@ -1,38 +1,69 @@
...
-mkdir -p $HOME/.gradle
-echo "checkstyle.ignoreFailures=false" >> $HOME/.gradle/gradle.properties
-echo "failOnWarnings=true" >> $HOME/.gradle/gradle.properties
-echo "errorProne=true" >> $HOME/.gradle/gradle.properties
+if [[ ! -e /tmp/protobuf ]]; then
+  ln -s $PROTO_INSTALL_DIR /tmp/protobuf;
+fi
+
+if [[ "$(readlink -f /tmp/protobuf)" != "$PROTO_INSTALL_DIR" ]]; then
+  exit 1;
```
Comment (ejona86) [inline buildscripts/kokoro/unix.sh]: echo an error

File: buildscripts/kokoro/unix.sh [hunk 5]
```diff
@@ -1,38 +1,69 @@
...
 pushd examples
-./gradlew build
+./gradlew build $GRADLE_FLAGS
 # --batch-mode reduces log spam
 mvn verify --batch-mode
 popd
 # TODO(zpencer): also build the GAE examples
+
+LOCAL_MVN_TEMP=$(mktemp -d)
+./gradlew clean grpc-compiler:build grpc-compiler:uploadArchives -PtargetArch=x86_64 -Dorg.gradle.parallel=false -PrepositoryDir=$LOCAL_MVN_TEMP $GRADLE_FLAGS
```
Comment (ejona86) [inline buildscripts/kokoro/unix.sh]: wrap line?

**Pull request merged**